temp_link_generator.php

```
@@ -0,0 +1,89 @@
+ * @return string The temporary link URL
+ */
+function generate_temp_link($bucket, $key, $access_key, $timestamp = null) {
+    global $server_url;
+    
+    if ($timestamp === null) {
+        $timestamp = time();
+    }
+    
+    // Generate the MD5 hash
+    $data = $timestamp . $bucket . $key . $access_key;
+    $link_key = md5($data);
```

PHPMD (2.15.0)

Avoid unused local variables such as '$link_key'. (Unused Code Rules)

(UnusedLocalVariable)